src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -183,8 +219,7 @@ private StorageListItem GetStorageListItem(IListBlobItem listBlobItem)
             string name = GetName(resourceUri);
 
             CloudBlockBlob blob = _directory.GetBlockBlobReference(name);
-
-            await blob.DeleteAsync(cancellationToken);
+            await blob.DeleteAsync(deleteSnapshotsOption:DeleteSnapshotsOption.IncludeSnapshots, accessCondition:null, options:null, operationContext:null,  cancellationToken:cancellationToken);
```

Comment:
On the delete operation of a blob, all snapshots will get deleted, is that the desired behavior? Essentially we don't have a recovery on delete.